src/core/net/nat64_translator.hpp

```diff
@@ -328,35 +339,63 @@ class Translator : public InstanceLocator, private NonCopyable
 
         typedef String<Ip6::Address::kInfoStringSize + Ip4::Address::kAddressStringSize + 4> InfoString;
 
-        void       Touch(TimeMilli aNow) { mExpiry = aNow + kAddressMappingIdleTimeoutMsec; }
+        struct ip6MapSearch
+        {
+            Ip6::Address ip6Addr;
+            uint16_t     srcPortOrId;
+        };
+        struct ip4MapSearch
+        {
+            Ip4::Address ip4Addr;
+            uint16_t     translatedPortOrId;
+        };
+
+        void       Touch(TimeMilli aNow, uint8_t aProtocol);
         InfoString ToString(void) const;
         void       CopyTo(otNat64AddressMapping &aMapping, TimeMilli aNow) const;
 
         uint64_t mId; // The unique id for a mapping session.
 
         Ip4::Address mIp4;
         Ip6::Address mIp6;
+        uint16_t     mSrcPortOrId;
+        uint16_t     mTranslatedPortOrId;
         TimeMilli    mExpiry; // The timestamp when this mapping expires, in milliseconds.
 
         ProtocolCounters mCounters;
 
     private:
         bool Matches(const Ip4::Address &aIp4) const { return mIp4 == aIp4; }
         bool Matches(const Ip6::Address &aIp6) const { return mIp6 == aIp6; }
+        bool Matches(const uint16_t aPort) const { return mTranslatedPortOrId == aPort; }
         bool Matches(const TimeMilli aNow) const { return mExpiry < aNow; }
 
+        bool Matches(const ip6MapSearch &aIp6Search) const
+        {
+            return ((mIp6 == aIp6Search.ip6Addr) && (mSrcPortOrId == aIp6Search.srcPortOrId));
+        }
+        bool Matches(const ip4MapSearch &aIp4Search) const
```

Comment:
At some point, I would like to extend the `FindMatching()` related methods to allow `Matches()` with multiple input parameters. This would remove the need to define these `struct` to search.

Reply:
Submitted this PR which should allow us to pass multiple input args in `FindMatching()` (for `LinkedList`): 
- https://github.com/openthread/openthread/pull/11065